docs/book/src/clusterctl/commands/alpha-topology-plan.md

```diff
@@ -1,3 +1,448 @@
+│   ├── DockerMachineTemplate_default_example-cluster-md-0-infra-qz9qk.yaml
+│   ├── KubeadmConfigTemplate_default_example-cluster-md-0-bootstrap-m29vz.yaml
+│   ├── KubeadmControlPlane_default_example-cluster-b2lhc.yaml
+│   ├── MachineDeployment_default_example-cluster-md-0-pqscg.yaml
+│   └── Secret_default_example-cluster-shim.yaml
+└── modified
+    ├── Cluster_default_example-cluster.diff
+    ├── Cluster_default_example-cluster.jsonpatch
+    ├── Cluster_default_example-cluster.modified.yaml
+    └── Cluster_default_example-cluster.original.yaml
+```
+
+### Plan changes to Cluster topology
+
+When making changes to a Cluster topology the `clusterctl alpha topology plan` can be used to analyse how the underlying objects will be affected.
+
+```shell
+clusterctl alpha topology plan -f modified-example-cluster.yaml -o output/
+```
+
+The `modified-example-cluster.yaml` scales up the control plane to 3 replicas and adds additional labels to the machine deployment.
+<details>
+<summary>View <code>modified-example-cluster.yaml</code></summary>
+
+```yaml
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: Cluster
+metadata:
+  name: "example-cluster"
+  namespace: default
+  labels:
+    cni: kindnet
+spec:
+  clusterNetwork:
+    services:
+      cidrBlocks: ["10.128.0.0/12"]
+    pods:
+      cidrBlocks: ["192.168.0.0/16"]
+    serviceDomain: "cluster.local"
+  topology:
+    class: example-cluster-class
+    version: v1.21.2
+    controlPlane:
+      metadata: {}
+      # Scale up the control plane from 1 -> 3.
+      replicas: 3
+    workers:
+      machineDeployments:
+      - class: "default-worker"
+        # Apply additional labels.
+        metadata: 
+          labels:
+            test-label: md-0-label
+        name: "md-0"
+        replicas: 1
+```
+</details>
+
+Produces an output similar to this:
+```shell
+Detected a cluster with Cluster API installed. Will use it to fetch missing objects.
+No ClusterClasses will be affected by the changes.
+The following Clusters will be affected by the changes:
+ ＊ default/example-cluster
+
+Changes for Cluster "default/example-cluster": 
+
+  NAMESPACE  KIND                 NAME                        ACTION    
+  default    KubeadmControlPlane  example-cluster-l7kx8       modified  
+  default    MachineDeployment    example-cluster-md-0-j58ln  modified  
+
+Modified objects are written to directory "output/modified"
+```
+
+### Rebase a Cluster to a different ClusterClass
+
+The command can be used to plan if a Cluster can be successfully rebased to a different ClusterClass.
+
+Rebasing a Cluster to a different ClusterClass:
+```shell
+# Rebasing from `example-cluster-class` to `another-cluster-class`.
+clusterctl alpha topology plan -f rebase-example-cluster.yaml -o output/
+```
+The `example-cluster` Cluster is rebased from `example-cluster-class` to `another-cluster-class`. In this example `another-cluster-class` is assumed to be available in the management cluster.
+
```

Comment:
Q: Should we provide the `another-cluster-class.yaml` here to?

Mainly asking because everything else was provided except that file.

Reply:
Did not provide that on purpose. `another-cluster-class` could be both a compatible and compatible cluster class. Hence I provided 2 sample output - one for compatible and another for incompatible.

Reply:
Okay, got it